internal/server/http/server.go

```diff
@@ -73,31 +72,30 @@ func PrepareAccessControl(config killgrave.ConfigCORS) (h []handlers.CORSOption)
 // Build read all the files on the impostersPath and add different
 // handlers for each imposter
 func (s *Server) Build() error {
-	if _, err := os.Stat(s.impostersPath); os.IsNotExist(err) {
-		return fmt.Errorf("%w: the directory %s doesn't exists", err, s.impostersPath)
+	done := make(chan struct{})
+	imposterFiles, errc := findImposters(s.impostersPath, done)
+	go s.processImposters(imposterFiles, done)
+
+	if err := <-errc; err != nil {
+		close(done)
+		return fmt.Errorf("%w: failed to find imposters", err)
 	}
-	var imposterFileCh = make(chan string)
-	var done = make(chan bool)
+	return nil
+}
 
-	go func() {
-		findImposters(s.impostersPath, imposterFileCh)
-		done <- true
-	}()
-	for {
+func (s *Server) processImposters(imposterFiles <-chan string, done <-chan struct{}) {
+	var imposters []Imposter
+	for f := range imposterFiles {
 		select {
-		case f := <-imposterFileCh:
-			var imposters []Imposter
-			err := s.unmarshalImposters(f, &imposters)
-			if err != nil {
+		case <-done:
+			return
+		default:
+			if err := s.unmarshalImposters(f, &imposters); err != nil {
 				log.Printf("error trying to load %s imposter: %v", f, err)
-			} else {
-				s.addImposterHandler(imposters, f)
-				log.Printf("imposter %s loaded\n", f)
+				continue
 			}
-		case <-done:
-			close(imposterFileCh)
-			close(done)
-			return nil
+			s.addImposterHandler(imposters, f)
```

Comment:
Checking the branch I see that there no test that pass for here, can you add it?